app/instrumentation-tests/src/org/commcare/androidTests/MenuTests.kt

```diff
@@ -0,0 +1,206 @@
+        InstrumentationUtility.gotoHome()
+
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText(homeMenuItems[3]))
+                .perform(click())
+        onView(withText("About CommCare"))
+                .check(matches(isDisplayed()))
+        onView(withText("OK"))
+                .perform(click())
+        InstrumentationUtility.gotoHome()
+
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText(homeMenuItems[4]))
+                .perform(click())
+        onView(withResourceName("action_bar_title"))
+                .check(matches(withText("CommCare > Advanced")))
+        checkStringExists(advancedOptions)
+        InstrumentationUtility.gotoHome()
+
+        InstrumentationUtility.openOptionsMenu()
+        onView(withText(homeMenuItems[5]))
+                .perform(click())
+        onView(withResourceName("action_bar_title"))
+                .check(matches(withText("CommCare > Settings")))
+        checkStringExists(arrayOf(
+                "Auto Update Frequency",
+                "Set Print Template",
+                "Grid Menus Enabled",
+                "Fuzzy Search Matches",
+                "Opt Out of Analytics"))
+        InstrumentationUtility.gotoHome()
+    }
+
+    @Test
+    fun testAdvancedActions() {
+        openAdvancedOption(0)
+        onView(withText("Do you want to send, receive, or submit forms?"))
+                .check(matches(isDisplayed()))
+        onView(withId(R.id.negative_button))
+                .perform(click())
+        Intents.intended(IntentMatchers.hasComponent(CommCareWiFiDirectActivity::class.java.name))
+        InstrumentationUtility.gotoHome()
+
+        openAdvancedOption(1)
+        val formDumpConfirmation = "Do not use this feature unless you have been trained to do so. Do you wish to proceed?"
+        onView(withText(formDumpConfirmation))
+                .check(matches(isDisplayed()))
+        onView(withId(R.id.negative_button))
+                .perform(click())
+        onView(withResourceName("action_bar_title"))
+                .check(matches(withText("CommCare > Advanced")))
+        onView(withText(advancedOptions[1]))
+                .perform(click())
+        onView(withText(formDumpConfirmation))
+                .check(matches(isDisplayed()))
+        onView(withId(R.id.positive_button))
+                .perform(click())
+        onView(withText(startsWith("Dump Forms")))
+                .check(matches(isDisplayed()))
+        onView(withText(startsWith("Submit Forms")))
+                .check(matches(isDisplayed()))
+        Espresso.pressBack()
+        onView(withText(formDumpConfirmation))
+                .check(doesNotExist())
+        InstrumentationUtility.gotoHome()
+
+        openAdvancedOption(4)
+        Intents.intended(IntentMatchers.hasComponent(CommCareVerificationActivity::class.java.name))
+        onView(withId(R.id.home_gridview_buttons))
```

Comment:
we can't check for toast messages as an indication of success ?

Reply:
No, unfortunately we can't with espresso.